### ISSUE TITLE
Test failures with pyshp 2.0.0

### ISSUE THREAD

**Issue by sebastic** [pytroll/pycoast#17]
#### Problem description

pycoast 1.0.0 fails to build with pyshp 2.0.0 due to test failures:
```
======================================================================
ERROR: test_dateline_boundary_cross (pycoast.tests.test_pycoast.TestPIL)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/pycoast-1.0.0+dfsg/pycoast/tests/test_pycoast.py", line 254, in test_dateline_boundary_cross
    lon_placement='b', lat_placement='lr')
  File "/build/pycoast-1.0.0+dfsg/pycoast/cw_pil.py", line 254, in add_grid
    lon_placement=lon_placement, lat_placement=lat_placement)
  File "/build/pycoast-1.0.0+dfsg/pycoast/cw_base.py", line 255, in _add_grid
    tmpshape = shapefile.Writer("")
  File "/usr/lib/python2.7/dist-packages/shapefile.py", line 1030, in __init__
    raise Exception('Either the target filepath, or any of shp, shx, or dbf must be set to create a shapefile.')
Exception: Either the target filepath, or any of shp, shx, or dbf must be set to create a shapefile.

======================================================================
ERROR: test_dateline_cross (pycoast.tests.test_pycoast.TestPIL)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/pycoast-1.0.0+dfsg/pycoast/tests/test_pycoast.py", line 228, in test_dateline_cross
    lon_placement='b', lat_placement='lr')
  File "/build/pycoast-1.0.0+dfsg/pycoast/cw_pil.py", line 254, in add_grid
    lon_placement=lon_placement, lat_placement=lat_placement)
  File "/build/pycoast-1.0.0+dfsg/pycoast/cw_base.py", line 255, in _add_grid
    tmpshape = shapefile.Writer("")
  File "/usr/lib/python2.7/dist-packages/shapefile.py", line 1030, in __init__
    raise Exception('Either the target filepath, or any of shp, shx, or dbf must be set to create a shapefile.')
Exception: Either the target filepath, or any of shp, shx, or dbf must be set to create a shapefile.

======================================================================
ERROR: test_grid (pycoast.tests.test_pycoast.TestPIL)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/pycoast-1.0.0+dfsg/pycoast/tests/test_pycoast.py", line 161, in test_grid
    outline='blue', minor_outline='blue')
  File "/build/pycoast-1.0.0+dfsg/pycoast/cw_pil.py", line 254, in add_grid
    lon_placement=lon_placement, lat_placement=lat_placement)
  File "/build/pycoast-1.0.0+dfsg/pycoast/cw_base.py", line 255, in _add_grid
    tmpshape = shapefile.Writer("")
  File "/usr/lib/python2.7/dist-packages/shapefile.py", line 1030, in __init__
    raise Exception('Either the target filepath, or any of shp, shx, or dbf must be set to create a shapefile.')
Exception: Either the target filepath, or any of shp, shx, or dbf must be set to create a shapefile.

======================================================================
ERROR: test_grid_file (pycoast.tests.test_pycoast.TestPIL)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/pycoast-1.0.0+dfsg/pycoast/tests/test_pycoast.py", line 203, in test_grid_file
    outline='blue', minor_outline='blue')
  File "/build/pycoast-1.0.0+dfsg/pycoast/cw_pil.py", line 289, in add_grid_to_file
    lon_placement=lon_placement, lat_placement=lat_placement)
  File "/build/pycoast-1.0.0+dfsg/pycoast/cw_pil.py", line 254, in add_grid
    lon_placement=lon_placement, lat_placement=lat_placement)
  File "/build/pycoast-1.0.0+dfsg/pycoast/cw_base.py", line 255, in _add_grid
    tmpshape = shapefile.Writer("")
  File "/usr/lib/python2.7/dist-packages/shapefile.py", line 1030, in __init__
    raise Exception('Either the target filepath, or any of shp, shx, or dbf must be set to create a shapefile.')
Exception: Either the target filepath, or any of shp, shx, or dbf must be set to create a shapefile.

======================================================================
ERROR: test_grid_geos (pycoast.tests.test_pycoast.TestPIL)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/pycoast-1.0.0+dfsg/pycoast/tests/test_pycoast.py", line 180, in test_grid_geos
    write_text=False)
  File "/build/pycoast-1.0.0+dfsg/pycoast/cw_pil.py", line 254, in add_grid
    lon_placement=lon_placement, lat_placement=lat_placement)
  File "/build/pycoast-1.0.0+dfsg/pycoast/cw_base.py", line 255, in _add_grid
    tmpshape = shapefile.Writer("")
  File "/usr/lib/python2.7/dist-packages/shapefile.py", line 1030, in __init__
    raise Exception('Either the target filepath, or any of shp, shx, or dbf must be set to create a shapefile.')
Exception: Either the target filepath, or any of shp, shx, or dbf must be set to create a shapefile.

======================================================================
ERROR: test_grid_nh (pycoast.tests.test_pycoast.TestPIL)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/pycoast-1.0.0+dfsg/pycoast/tests/test_pycoast.py", line 279, in test_grid_nh
    lon_placement='tblr', lat_placement='')
  File "/build/pycoast-1.0.0+dfsg/pycoast/cw_pil.py", line 254, in add_grid
    lon_placement=lon_placement, lat_placement=lat_placement)
  File "/build/pycoast-1.0.0+dfsg/pycoast/cw_base.py", line 255, in _add_grid
    tmpshape = shapefile.Writer("")
  File "/usr/lib/python2.7/dist-packages/shapefile.py", line 1030, in __init__
    raise Exception('Either the target filepath, or any of shp, shx, or dbf must be set to create a shapefile.')
Exception: Either the target filepath, or any of shp, shx, or dbf must be set to create a shapefile.

----------------------------------------------------------------------
Ran 11 tests in 14.555s

FAILED (errors=6, skipped=1)
```

Refer to the pyshp changelog for possible causes:

 https://github.com/GeospatialPython/pyshp/tree/2.0.0#200

**Comment by mraspaud** [comment]
Thanks for reporting this.

**Comment by djhoese** [comment]
So it looks like these errors are just from taking shortcuts and creating an empty writer object. Now the writer objects need an actual filename. I would guess that if we just put a temporary filename in their these would be fixed.

**Comment by djhoese** [comment]
This has been fixed in current master with #19. I'll try to release something in the next week.
